### PR TITLE
Validate terminateAfter before insert into DB

### PR DESCRIPTION
**Pull request by anniezhang77**
risk: low

### CODE REVIEW

File: orchard-core/src/main/scala/com/salesforce/mce/orchard/db/OrchardDatabase.scala [hunk 1]
```diff
@@ -51,6 +51,10 @@ class OrchardDatabase(conf: Config) {
       }
 
     val addResources = ResourceTable() ++= workflow.resources.map { r =>
+      assert(
+        r.terminateAfter > 0D && r.terminateAfter <= 8D,
```
Comment (realstraw) [inline orchard-core/src/main/scala/com/salesforce/mce/orchard/db/OrchardDatabase.scala:55]: As a hard limit, i think 8 is too tight. let's make it 24*30 instead

**Comment by realstraw** [comment]
while it's good to protect the database, the actual ask is to make sure to validate the JSON payload in the `POST` workflow request, and return Bad Request from the webservice controller, so that client would get the failure feedback immediately.

File: orchard-ws/app/controllers/WorkflowController.scala [hunk 2]
```diff
@@ -73,6 +73,30 @@ class WorkflowController @Inject() (
...
+      } yield resource.maxAttempt
+
+    if (invalidTerminateAt.nonEmpty) {
+      Left("Invalid terminateAt value")
+    } else if (invalidMaxAttempts.nonEmpty) {
+      Left("Invalid maxAttempt value")
+    } else {
+      Right(request)
+    }
+  }
```
Comment (anniezhang77) [inline orchard-ws/app/controllers/WorkflowController.scala:98]: @realstraw don't know what is a better way of doing this. Feel free to comment for improvements.

Comment (realstraw) [reply orchard-ws/app/controllers/WorkflowController.scala:98]: can probably just call this in `processWorkflowRequest` instead and make it return `Future[Either]`.

Also, it'd be good to make the error message containing more detail. e.g. "Invalid terminateAt value, must be a number between 0 and 24 * 30"`

Comment (anniezhang77) [reply orchard-ws/app/controllers/WorkflowController.scala:98]: you mean make `processWorkflowRequest` return `Either[String, Future]` right?

Comment (anniezhang77) [reply orchard-ws/app/controllers/WorkflowController.scala:98]: hmm either way is kind of awkward.

Comment (anniezhang77) [reply orchard-ws/app/controllers/WorkflowController.scala:105]: I think the previous version is cleaner

Comment (realstraw) [reply orchard-ws/app/controllers/WorkflowController.scala:105]: ok, let's use the previous version then

Comment (anniezhang77) [reply orchard-ws/app/controllers/WorkflowController.scala:97]: okay, reverted, added more detailed error messages

Comment (realstraw) [reply orchard-ws/app/controllers/WorkflowController.scala:97]: thanks!

**Review by realstraw — APPROVED** [review]

**Pull request merged**